step-controller/step-controller-server/src/main/java/step/core/scheduler/ExecutionJob.java

```diff
@@ -18,25 +18,87 @@
  ******************************************************************************/
 package step.core.scheduler;
 
+import org.bson.types.ObjectId;
 import org.quartz.Job;
 import org.quartz.JobExecutionContext;
 import org.quartz.JobExecutionException;
 
+import step.automation.packages.AutomationPackageEntity;
+import step.automation.packages.AutomationPackageLocks;
+import step.core.controller.ControllerSetting;
+import step.core.controller.ControllerSettingAccessor;
 import step.core.execution.ExecutionEngine;
 
+import static step.automation.packages.AutomationPackageLocks.AUTOMATION_PACKAGE_READ_LOCK_TIMEOUT_SECS;
+import static step.automation.packages.AutomationPackageLocks.AUTOMATION_PACKAGE_READ_LOCK_TIMEOUT_SECS_DEFAULT;
+
 public class ExecutionJob implements Job {
 	
 	private final ExecutionEngine executionEngine;
+	private final ExecutionTaskAccessor executionTaskAccessor;
+	private final ControllerSettingAccessor controllerSettingAccessor;
+	private final AutomationPackageLocks automationPackageLocks;
 	private final String executionId;
-	
-	public ExecutionJob(ExecutionEngine executionEngine, String executionId) {
-		super();
+	private final String executionTaskID;
+
+	public ExecutionJob(ExecutionEngine executionEngine, ExecutionTaskAccessor executionTaskAccessor, ControllerSettingAccessor controllerSettingAccessor, AutomationPackageLocks lock, String executionId, String executionTaskID) {
 		this.executionEngine = executionEngine;

```

Comment:
You may want to add `Objects.requireNonNull` around the assignments of parameters which are expected to be non-null. This way, should there ever come in a "wrongly null" value, an exception is thrown early instead of "somewhere else" at runtime. It also serves as a sort of "in-code" documentation concerning what can be expected to be null, and what not.